File: sum-of-two-integers.cpp
class Solution {
public:
  int getSum(int a, int b) {
    int total = a, carry = b;
    while (carry) {
      int new_total = total ^ carry;
      int new_carry = total & carry;
      total = new_total;
      carry = new_carry << 1;
    }
    return total;
  }
};

File: sum-of-two-integers.py
class Solution:
    def getSum(self, a: int, b: int) -> int:
        mask = 0xFF_FF_FF_FFF
        kMax = 2000

        while b != 0:
            a, b = (a ^ b) & mask, ((a & b) << 1) & mask

        return a if a < kMax else ~(a ^ mask)
